include/exadg/incompressible_navier_stokes/user_interface/parameters.cpp

```diff
@@ -412,7 +412,7 @@ Parameters::check(dealii::ConditionalOStream const & pcout) const
     }
   }
 
-  if(viscosity_is_variable() and nonlinear_problem_has_to_be_solved())
+  if(nonlinear_viscous_problem())
```

Comment:
This new version is in my opinion equivalent to the old code, just simpler. However, I am not sure if this is what we want!? Imagine we have a variable viscosity problem, but the variable viscosity is treated explicitly. In that case, we would currently allow over-integration for the momentum operator if `param.quad_rule_linearization` is set accordingly. Operators are then initialized with different quad-rules, which does not seem to be supported.

@richardschu If you agree, I would change the code accordingly in the present PR, i.e. write just `if(viscosity_is_variable())` and thereby broaden the cases where this Assert is checked.

Reply:
In this comment, I consider the case of an implicit convective problem and an explicitly treated variable viscosity. The Code in `SpatialOperatorBase`  would then assign an over-integration quad-rule for the momentum operator for both the old code version and the new version suggested in this PR.

Reply:
I decided to change the if-statement as described above for now.

Reply:
I did not use over-integration for non-Newtonian fluids yet and I am not sure how the experiences from turbulent flow (only? idk!) transfer to the cases where the non-Newtonian models make most sense (much lower Re, usually). This is why I did not extend towards over-integration with variable viscosity.

Extend the `if` and broaden the cases this is checked, but please use
`if(viscous_problem() and viscosity_is_variable())`
since we might not have a viscous term, but just the `turbulence_model_data.is_active` or `generalized_newtonian_model_data.is_active` set wrong in an application code.

Reply:
yes, above I forgot `viscous_problem()`, but the current code should have that if-statement already.